lib/mix/tasks/phx.gen.socket.ex

```diff
@@ -0,0 +1,126 @@
+defmodule Mix.Tasks.Phx.Gen.Socket do
+  @shortdoc "Generates a Phoenix socket handler"
+
+  @moduledoc """
+  Generates a Phoenix socket handler.
+
+      mix phx.gen.socket User
+
+  Accepts the module name for the socket
+
+  The generated files will contain:
+
+  For a regular application:
+
+    * a client in `assets/js`
+    * a socket in `lib/my_app_web/channels`
+    * a socket test in `test/my_app_web/channels`
+
+  For an umbrella application:
+
+    * a client in `apps/my_app_web/assets/js`
+    * a socket in `apps/my_app_web/lib/app_name_web/channels`
+    * a socket test in `apps/my_app_web/test/my_app_web/channels`
+
+  """
+  use Mix.Task
+
+  @doc false
+  def run(args) do
+    if Mix.Project.umbrella?() do
+      Mix.raise(
+        "mix phx.gen.socket must be invoked from within your *_web application root directory"
+      )
+    end
+
+    [socket_name, pre_existing_channel] = validate_args!(args)
```

Comment:
Can we pass the channel under an option instead? Something like `--from-channel Room`. I want to avoid passing the channel as accident... unless we want to make it the channel argument a public API.

Reply:
I changed this locally!